src/javascript/app/base/header.js

```diff
@@ -441,7 +441,9 @@ const Header = (() => {
         applyToAllElements('.language-menu-item', (el) => {
             el.addEventListener('click', () => {
                 const item_language = el.getAttribute('id');
-                if (item_language === current_language) return;
+                if (item_language !== current_language) SocketCache.clear();
+                else return;
```

Comment:
I think we can improve the readability of this section by adding the cache clearing method after the previous `if` without any need to add an `else`:

```suggestion
                if (item_language === current_language) return;
                SocketCache.clear()
```